modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SidebarPanel.js

```diff
@@ -14,26 +14,86 @@
 
 import ClayAlert from '@clayui/alert';
 import ClayLoadingIndicator from '@clayui/loading-indicator';
-import {useTimeout} from 'frontend-js-react-web';
+import {useIsMounted, useTimeout} from 'frontend-js-react-web';
 import {fetch} from 'frontend-js-web';
-import React, {useEffect, useImperativeHandle, useRef, useState} from 'react';
+import React, {useEffect, useImperativeHandle, useReducer, useRef} from 'react';
 
 import Sidebar from './components/Sidebar';
 
+const initialState = {
+	data: null,
+	error: null,
+	html: null,
+	loading: false,
+	open: true,
+};
+
 const SidebarPanel = React.forwardRef(
 	({fetchURL, onClose, viewComponent: View}, ref) => {
 		const CurrentView = useRef(View);
 
-		const [error, setError] = useState();
-		const [isLoading, setIsLoading] = useState();
-		const [isOpen, setIsOpen] = useState(true);
-		const [resourceData, setResourceData] = useState();
-
 		const delay = useTimeout();
 
+		const isMounted = useIsMounted();
+
+		const dataReducer = (state, action) => {
+			if (isMounted()) {
+				switch (action.type) {
+					case 'CLOSE_SIDEBAR':
+						return {
+							...state,
+							isOpen: false,
+						};
+
+					case 'LOAD_DATA':
+						return {
+							loading: true,
+						};
+
+					case 'OPEN_SIDEBAR':
+						return {
+							...state,
+							isOpen: true,
+						};
+
+					case 'SET_ERROR':
+						return {
+							error: action.error,
+							loading: false,
+						};
+
+					case 'SET_HTML':
+						return {
+							data: {
+								html: action.html,
+							},
+							error: null,
+							loading: false,
+						};
+
+					case 'SET_JSON':
+						return {
+							data: action.data,
+							error: action.data?.error,
+							loading: false,
+						};
+
+					default:
+						return initialState;
+				}
+			}
+		};
+
+		const [state, dispatch] = useReducer(dataReducer, initialState);
+
+		// Force 300 ms of waiting to render the response so loading
+		// looks more natural.
+
+		const dispatchWithDelay = (action) =>
+			delay(() => dispatch(action), 300);
```

Comment:
I know this isn't new behavior, but can you help me understand how you arrived at this 300ms delay? Seeing as we will already have some variable delay coming from the `fetch`, and then this happens on top of that.

Reply:
Just visual taste. Without the delay the change was inmediate or it looked like it was flickering, so the team wanted a little loading to happen so the user feel that something occurred. 500 was too much, 300 just felt enough 🤷

Reply:
Did you try simulating worse network conditions, like 3G? Maybe the effect is caused by `localhost` and there's enough delay in a real life scenario where making the user wait for an additional 300ms is unwarranted?

Maybe a fast fade-in/fade-out transition could solve the issue?

Reply:
Checked with Mid-tier mobile connection without the delay and looks ok. Removing it, thx.

Reply:
The variability concerns me; consider these two cases on a fast and slow network respectively:

- Fast network: server response comes back in 200ms, we add 300ms to that, and user ends up waiting 500ms; experience "feels good".
- Slow network: server response comes back in 1200ms, we add 300ms to that (needlessly), and user ends up waiting 1500ms overall. Experience sucks more than it should.

As far as UX goes I would much rather the flickering be tackle in another way (ie. close-to-open transition could coincide with fetch, show same-sized placeholder until data ready etc) but that is the teams choice. But given that they want an artificial delay, I think you should run the two async jobs concurrently, and only impose a delay if the fetch is too fast to add up to the desired interval. (Having said that, I see that your more recently pushed changes don't have the delay anymore, so I guess all this is stale.)